src/ucp/core/ucp_context.h

```diff
@@ -457,6 +463,13 @@ ucp_memory_type_detect(ucp_context_h context, const void *address, size_t length
     }
 
     return ucp_memory_type_detect_mds(context, address, length);
+
+not_found:
+    if (ucs_unlikely((length > 0) && !ucs_arch_is_mem_accessible(address))) {
```

Comment:
isn't incorrect for CUDA MANAGED and ROCM memories which are mem accessible and not segfault?
It true that even currently in static linking case, it is silently treating MANAGED and ROCM allocation as HOST memory, and which can cause incorrect data validation.

Reply:
shouldn't it be just lower performance? 
in any case if memtype cache doesn't work they would be assumed as HOST..

Reply:
actually it is a correctness issue. AFAIK, RDMA on MANAGED memory is invalid because pages might be migrated/copied to GPU while HCA accessing them. @Akshay-Venkatesh ?  I think we have a broken CUDA MANAGED memory support today after https://github.com/openucx/ucx/pull/4473 . need to revisit this.

below, I'm stating different behaviors for static linking case

w/o this PR:
```
1.  CUDA:  segfault
2 . CUDA Managed:  Managed memory assumed as HOST
3.  CUDA + CUDA Managed (first txfer is with CUDA): segfault
4.  CUDA Managed + CUDA (first txfer is with MANAGED): segfault
```

with PR: 

```
1. CUDA:  works
2. CUDA Managed:  Managed memory assumed as HOST
3. CUDA + CUDA Managed (first txfer is with CUDA):  works,  future Managed memory is correctly detected after cache detroyed with segv
4. CUDA Managed + CUDA (first tcfer is with MANAGED):  Managed memory treated as HOST until SEGV. then detected as MANAGED.

```

Lets assume we will have different protocols for HOST and MANAGED.
cases (1) & (3) cases are fixed with PR
case (2)  behavior don't change
case (4). user might have get it correct by disabling the cache after segfault w/o this PR.

Reply:
+1 to what Devendar mentioned. 

Page migrations are in the hands of cuda-runtime and pages can move while the NIC is in the process of transferring contents. Bottom-line we can't assume safety of registering managed memory buffers with IB and treat it like CPU buffers (only known exception to this is on Summit where reg_mr on managed memory buffers is legal).